providers/provider_default.go

```diff
@@ -116,6 +116,13 @@ func (p *ProviderData) EnrichSession(_ context.Context, _ *sessions.SessionState
 // Authorize performs global authorization on an authenticated session.
 // This is not used for fine-grained per route authorization rules.
 func (p *ProviderData) Authorize(_ context.Context, s *sessions.SessionState) (bool, error) {
+	if len(p.AllowedACRs) > 0 {
+		var _, ok = p.AllowedACRs[s.Acr]
+		if !ok {
+			return false, nil
+		}
+	}
+
```

Comment:
IMO testing of other code is NICE TO HAVE but this one is the purpose of your PR.

Plug your tests (1 success, one failure) like for groups in `providers/provider_default_test.go:73`